### PR TITLE
added stix custom registry for interop

### PR DESCRIPTION
**Pull request by jweissm**

### CODE REVIEW

**Review by clenk — CHANGES_REQUESTED** [review]
1. ~Does attack-pattern need to have `'kill_chain_phases'` in its `properties` block?~ No, because `kill_chain_phases` is in attack-pattern, not common-properties. Since common-properties is imported, requiring any previously-optional common properties would give errors.
2. Grouping doesn't require `identities_class`, as that is not one of its properties.
3. Could we put `created_by_ref` being required in the `common-properties` block? Then we could remove the entries for object types where that is the only new required property (Campaign, Course of Action, Grouping, Incident, Identity, Note, Opinion). This would also mean we would not need to include `created_by_ref` in the `required` block under the other object types, either, nor need to redefine it in their `properties` block.
4. I don't think we need to redefine `type` under `properties` for each object. Are there errors if we take that out?
5. `identity_class` is not required on Identity objects.
6. ~Infrastructure, intrusion set, location, malware, malware-analysis, observed-data, report, threat-actor, and tool all have properties that are in their `required` fields that aren't redefined in their `properties` blocks. Won't that give errors?~ No, because those properties are defined in those objects' properties blocks in the standard registry, not in common-properties.

File: stix2generator/language/interop_custom.json [hunk 1]
```diff
@@ -0,0 +1,309 @@
+{
```
Comment (clenk) [inline stix2generator/language/interop_custom.json]: Is there a reason this file is under `/language`? Maybe it should be one directory up, next to the default `stix21_registry.json`.

**Review by clenk — COMMENTED** [review]
Would also be cool to have an `--interop` commandline flag so users don't have to use `--extra-specs` and find the correct path.

File: stix2generator/language/interop_custom.json [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+                "type": "string",
+                "semantics": "stix-id",
+                "stix-type": "identity"
+            }
+        }
+    },
+    "malware": {
+        "type": "object",
+        "import": "common-properties",
+        "required": ["malware_types", "is_family", "first_seen", "last_seen", "architecture_execution_envs", "implementation_languages", "capabilities", "name", "created_by_ref"],
```
Comment (clenk) [inline stix2generator/language/interop_custom.json]: `name` and `is_family` are already required by the standard registry.

File: stix2generator/language/interop_custom.json [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+                "type": "string",
+                "semantics": "stix-id",
+                "stix-type": "identity"
+            }
+        }
+    },
+    "intrusion-set": {
+        "type": "object",
+        "import": "common-properties",
+        "required": ["resource_level", "primary_motivations", "created_by_ref"],
```
Comment (clenk) [inline stix2generator/language/interop_custom.json]:
```suggestion
        "required": ["resource_level", "primary_motivation", "created_by_ref"],
```
not plural

Comment (jweissm) [reply stix2generator/language/interop_custom.json]: fixed

File: stix2generator/language/interop_custom.json (hunk 2 above)
Comment (jweissm) [reply stix2generator/language/interop_custom.json]: removed

File: stix2generator/language/interop_custom.json (hunk 1 above)
Comment (jweissm) [reply stix2generator/language/interop_custom.json]: moved up on directory

File: stix2generator/interop_custom.json [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    },
+    "malware": {
+        "type": "object",
+        "import": "common-properties",
+        "required": ["malware_types", "first_seen", "last_seen", "architecture_execution_envs", "implementation_languages", "capabilities"]
+    },
+    "malware-analysis": {
+        "type": "object",
+        "import": "common-properties",
+        "required": ["version", "submitted", "analysis_started", "analysis_ended", "results"]
```
Comment (clenk) [inline stix2generator/interop_custom.json]:
```suggestion
        "required": ["version", "submitted", "analysis_started", "analysis_ended", "result"]
```

File: stix2generator/interop_custom.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    },
+    "location": {
+        "type": "object",
+        "import": "common-properties",
+        "required": ["region"]
+    },
+    "malware": {
+        "type": "object",
+        "import": "common-properties",
+        "required": ["malware_types", "first_seen", "last_seen", "architecture_execution_envs", "implementation_languages", "capabilities"]
```
Comment (clenk) [inline stix2generator/interop_custom.json]: These properties are still required for interop:
```suggestion
        "required": ["is_family", "name", "malware_types", "first_seen", "last_seen", "architecture_execution_envs", "implementation_languages", "capabilities"]
```

**Review by clenk — CHANGES_REQUESTED** [review]

**Review by clenk — APPROVED** [review]
Looks good to me now!

**Pull request merged**